pycargoebuild/__main__.py

```diff
@@ -71,13 +71,24 @@ def main(prog_name: str, *argv: str) -> int:
     load_license_mapping(args.license_mapping)
     args.license_mapping.close()
 
+    def locate_workspace_root(directory: Path) -> Path:
+        for current_dir in (directory.resolve() / "Cargo.lock" ).parents:
+            try:
+                with open(current_dir / "Cargo.lock", "rb"):
+                    return current_dir
```

Comment:
I actually meant returning it, i.e.:

```suggestion
                return open(current_dir / "Cargo.lock", "rb")
```

Then you should be able to call the function within `with`.

Reply:
I see, done.